src/Akka.Persistence.Sql/Extensions/ConnectionFactoryExtensions.cs

```diff
@@ -44,6 +47,24 @@ public static async Task ExecuteWithTransactionAsync(
             }
         }
 
+        public static async Task<T> ExecuteQueryWithTransactionAsync<T>(
+            this AkkaPersistenceDataConnectionFactory factory,
+            IActorRef queryPermitter,
+            IsolationLevel level,
+            CancellationToken token,
+            Func<AkkaDataConnection, CancellationToken, Task<T>> handler)
+        {
+            await queryPermitter.Ask<QueryStartGranted>(RequestQueryStart.Instance);
+            try
+            {
+                return await factory.ExecuteWithTransactionAsync(level, token, handler);
+            }
+            finally
+            {
+                queryPermitter.Tell(ReturnQueryStart.Instance);
+            }
```

Comment:
This is the query throttling mechanism, we always ask for a permission before we execute any DB operations